### ISSUE TITLE
[CLOSED] login_call_resetPassword

### ISSUE THREAD

**Issue by ulfgebhardt** [Zahlungsmittel/Zahlungsmittel#1189]
<a href="https://github.com/ulfgebhardt"><img src="https://avatars.githubusercontent.com/u/1238238?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ulfgebhardt](https://github.com/ulfgebhardt)**
_Tuesday Nov 23, 2021 at 02:31 GMT_
_Originally opened as https://github.com/gradido/gradido/pull/1130_

----

## 🍰 Pullrequest

Implement the existing calls:
- resetpassword
- checkEmail
- loginViaEmailVerificationCode
on Apollo.

Multiple processes will be unified and simplified:
- On Register -> send EMail -> set Password & Activate -> Login
- On Forgot Password -> send EMail -> set Password -> Login
- On NoPassword but Activated -> send Email -> set Password & Activate -> Login (legacy)

Following Calls will be implemented:
- `requestSetPassword`
- `setPassword`

Furthermore the two-step-call process will be a one-step-call process.
> Before: EMail Link -> Call for Login on open page-> Call for Reset Password/Activate Account on submit
> After: EMail Link -> Call for Reset Password/Activate Account on submit

### Issues

- fixes #836  
- fixes #837 
- fixes #833 
- includes #1131

### Todo

- [x] remove `checkEmail` & `loginViaEmailVerificationCode` calls
- [x] combine `checkEmail` & `loginViaEmailVerificationCode` with `resetPassword` calls into single `setPassword` call
- [x] integrate with `createUser` process
- [x] integrate with `loginUser` process for legacy reasons & not activated
- [ ] frontend implementations @Elweyn 

----

_**[ulfgebhardt](https://github.com/ulfgebhardt)** included the following code: https://github.com/gradido/gradido/pull/1130/commits_

**Comment by ulfgebhardt** [comment]
<a href="https://github.com/Elweyn"><img src="https://avatars.githubusercontent.com/u/33051975?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Elweyn](https://github.com/Elweyn)**
_Tuesday Nov 23, 2021 at 07:18 GMT_

----

frontend implementations TODOs
- [x] Withdraw password fields of registration form
- [x] Make a view that will handle (set password, Forgot password & loginViaEmailVerificationCode) same as thx.vue

**Comment by ulfgebhardt** [comment]
<a href="https://github.com/ulfgebhardt"><img src="https://avatars.githubusercontent.com/u/1238238?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ulfgebhardt](https://github.com/ulfgebhardt)**
_Friday Nov 26, 2021 at 02:33 GMT_

----

![image](https://user-images.githubusercontent.com/1238238/143517956-9ec45f93-3b23-4e13-a729-93e2a4bdda9b.png)

i skipped the whole resetPassword test since it completely changed - I think we need to rewrite it once we done with this

**Comment by ulfgebhardt** [comment]
<a href="https://github.com/ulfgebhardt"><img src="https://avatars.githubusercontent.com/u/1238238?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ulfgebhardt](https://github.com/ulfgebhardt)**
_Monday Dec 06, 2021 at 23:39 GMT_

----

I'd like to merge this and do the frontend part in another PR to reduce complexity. This means we reduce coverage requirement for the frontend to 84% again, since the whole check/reset-Password Component has to change and therefore no test is working anylonger on that. Whats your verdict @Mogge ?

**Comment by ulfgebhardt** [comment]
<a href="https://github.com/Elweyn"><img src="https://avatars.githubusercontent.com/u/33051975?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Elweyn](https://github.com/Elweyn)**
_Tuesday Dec 07, 2021 at 09:22 GMT_

----

@ulfgebhardt I changed the frontend on this branch!
- In case a User has not activated his email/account and try to login => get's redirected to /reset/login that shows a message to check his email to activate his account!
- In case a User sends a setPassword and get's an optin expired => Users get's redirected to the resetPassword page with a new message that says optin has expired want a new?
